mailpile/plugins/contacts.py

```diff
@@ -1161,7 +1161,9 @@ def _update_vcard_from_post(self, vcard, state=None):
                     'label': False,
                     'display': 'invisible'
                 })
-                tags[vcard.tag].slug = 'account-%d' % len(tags)
+                from mailpile.plugins.tags import Slugify
+                tags[vcard.tag].slug = Slugify(
+                    'account-%s' % vcard.email, tags=self.session.config.tags)
```

Comment:
I guess email is the best you've got, no guarantee of having a contact name in the vcard I suppose?